chat-widget/src/components/draggable/DraggableChatWidget.tsx

```diff
@@ -0,0 +1,171 @@
+            offsetLeft = window.innerWidth - positionRelativeToViewport.width - delta.left;
+        }
+
+        if (positionRelativeToViewport.y + positionRelativeToViewport.height > window.innerHeight) {
+            offsetTop = window.innerHeight - positionRelativeToViewport.height - delta.top;
+        }
+
+        repositionElement(draggableElement as HTMLElement, offsetLeft, offsetTop);
+        setPosition({offsetLeft, offsetTop});
+    }, []);
+
+    const resetPosition = useCallback((targetPosition: IDraggableElementPosition) => {
+        calculateOffsetsWithinViewport(props.elementId, targetPosition, delta); // Ensure viewport restriction
+    }, [delta]);
+
+    useEffect(() => {
+        if (props.disabled === true) {
+            return;
+        }
+
+        const cacheInitialPosition = () => {
+            const draggableElement: HTMLElement | null = document.getElementById(props.elementId);
+            const offsetLeft = (draggableElement as HTMLElement).offsetLeft as number;
+            const offsetTop = (draggableElement as HTMLElement).offsetTop as number;
+            setInitialPosition({offsetLeft, offsetTop});
+        };
+
+        const calculateOffsets = () => {
+            const draggableElement: HTMLElement | null = document.getElementById(props.elementId);
+            const offsetLeft = (draggableElement as HTMLElement).offsetLeft as number;
+            const offsetTop = (draggableElement as HTMLElement).offsetTop as number;
+
+            // Calculates the delta between the position of the widget and the position of the widget relative to the viewport which will be used for repositioning
+            const positionRelativeToViewport = (draggableElement as HTMLElement).getBoundingClientRect();
+            const left = positionRelativeToViewport.left - offsetLeft;
+            const top = positionRelativeToViewport.top - offsetTop;
+            setDelta({left, top});
+
+            calculateOffsetsWithinViewport(props.elementId, {offsetLeft, offsetTop}, {left, top});
+        };
+
+        calculateOffsets();
+        cacheInitialPosition();
+
+        window.addEventListener("resize", calculateOffsets);
+
+        return () => {
+            window.removeEventListener("resize", calculateOffsets);
+        };
+    }, [props.disabled]);
+
+    useEffect(() => {
+        if (props.disabled === true) {
+            return;
+        }
+
+        if (state.appStates.conversationState == ConversationState.Closed) {
+            resetPosition(initialPosition);
+        } else if (state.appStates.isMinimized) {
+            const draggableElement: HTMLElement | null = document.getElementById(props.elementId);
+            const offsetLeft = (draggableElement as HTMLElement).offsetLeft as number;
+            const offsetTop = (draggableElement as HTMLElement).offsetTop as number;
+
+            if (!cachedPosition) {
+                setCachedPosition({offsetLeft, offsetTop});

```

Comment:
When we maximize the chat widget after it has been minimized, do we expect the maximized location to be where the chat widget was, right before minimization?

Reply:
Yes, that's correct